server.js

```diff
@@ -1,33 +1,165 @@
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+
+import spotifyData from "./data/spotify-releases.json";
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(bodyParser.json());
+
+// error message
+const ERROR_RELEASES_NOT_FOUND = { error: "No release matched your request" };
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send("Welcome to the Spotify Releases API");
+});
+
+// This route will return a collection of releases. It can be filtered by type, artist and title.
+app.get("/releases", (req, res) => {
+  const { type, artist, title } = req.query;
+
+  let filteredReleases = spotifyData;
+
+  if (type) {
+    filteredReleases = filteredReleases.filter(
+      (item) => item.album_type === type
+    );
+  };
+  if (artist) {
+    filteredReleases = filteredReleases.filter(
+      (item) => item.artists.map((artist) => artist.name).toString().includes(artist)
+    );
+  };
+  if (title) {
+    filteredReleases = filteredReleases.filter(
+      (item) => item.name && item.name.toString().includes(title)
+    );
+  };
+  if (filteredReleases.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: filteredReleases.length,
+      releases: filteredReleases
+    });
+  }
+});
+
+// This route will return a single release based on id
+app.get("/releases/:id", (req, res) => {
+  const { id } = req.params;
+  const release = spotifyData.find(
+    (item) => item.id === id
+  );
+  if (!release) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json(release);
+  }
+});
+
+// This route will return a collection of releases for the specified artist
+app.get("/releases/artist/:artist", (req, res) => {
+  const { artist } = req.params;
+  const filteredReleases = spotifyData.filter(
+    (item) => item.artists.map((artist) => artist.name).toString() === artist
+  );
+  if (filteredReleases.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: filteredReleases.length,
+      releases: filteredReleases
+    });
+  };
+});
```

Comment:
Endpoints should be named after what they return and not how they are filtered. This endpoint could be called `artists` (in plural) if that's what you're returning, but since you're returning releases, this endpoint is redundant - since you already have an `artist` query under the `/releases` endpoint :)